Added Licensing documentation (#179)
Mirantis/mke-docs: content/docs/getting-started/licensing-mke4.md

---
title: Licensing MKE4
weight: 2
---

{{< callout type="warning" >}}

You must have a valid license to lawfully run MKE4. For more
information, refer to [Mirantis Agreements and Terms](https://legal.mirantis.com/).

{{< /callout >}}

## Obtain your MKE4 license

> [Install the MKE image](install-MKE-CLI.md) prior to downloading your MKE4 license.

Reviewer: ```suggestion
[Install the MKE CLI](install-MKE-CLI.md) prior to downloading your MKE4 license.
```

Using the word "Image" is not correct here, MKE CLI is a binary, not an image. The correct options would be
- mkectl binary
- MKE CLI
- MKE CLI binary

> prior to downloading your MKE4 license

Also, not sure why it has to be done prior to getting a license. Not necessarily. Users can get their license first and then proceed with installing MKE 4: getting the binary, configuring and deploying their cluster.

> [Install the MKE CLI](install-MKE-CLI.md) prior to downloading your MKE4 license.
> 1. Locate the Welcome to Mirantis' CloudCare Portal email sent to you from Mirantis

Reviewer 2: ```suggestion
[Install the MKE CLI](install-MKE-CLI.md) prior to downloading your MKE4 license.

1. Locate the Welcome to Mirantis' CloudCare Portal email sent to you from Mirantis
```

>    Support. If you do not have the email, confirm with your Designated Administrator
>    that you have added as a Designated Contact.

Reviewer 2: ```suggestion
   Support. If you do not have the email, confirm with your Designated Administrator
   that you have been added as a Designated Contact.
```

2. Click **Environments** in the top navigation bar of the MKE web UI.

3. Click the **Cloud Name** that is associated with the license you want to download.

4. Scroll down to **License Information** and click the **License File URL**. 
   A new tab opens in your browser.

> 5. Click **View file** to download your license file.

Reviewer: I don't remember seeing these steps in MKE 3 docs, so, I cannot say if this is the correct way or not. Maybe someone who is more familiar with the process of getting license should also review this.

Reviewer: oh I just realized that these are actually on the MKE 3 docs. I don't know how I never saw them. I thought you got it from somewhere else, but since we have it in MKE 3, and no one complained, I think we should be good

{{< callout type="info" >}}

Though MKE4 is generally a subscription-only service, you can obtain a free trial license from Mirantis. Make your request using the [Mirantis contact form](https://www.mirantis.com/contact).

{{< /callout >}}

## Add the license

1. Log in to the MKE web UI with an administrator account.
2. In the left-side navigation panel, navigate to **Admin Settings** -> **License**.
3. Insert the license into the **license key** field. To do this, click
   **Choose File** and navigate to the ``.lic`` file you previously downloaded.
   Alternately, you can copy-paste the contents of the ``.lic`` file into the
   field as a text string.

   ![Add a license](img/add-a-license.png)

4. Click **Save settings** to update the MKE4 license.

## Set the license in the configuration

> 1. Set the license under ``spec.license.token`` in the MKE4

Reviewer 2: ```suggestion
1. Insert the license into ``spec.license.token`` in the MKE4
```

   configuration file:

    ```yaml
    spec:
      license:
        token: <your-license-file>
    ```

2. Apply the license:

   ```commandline
   mkectl apply
   ```

3. Check the license status:

   ```commandline
   kubectl -n mke get mkeconfig mke -ojsonpath="{.status.licenseStatus}" | jq 
   ```

   Example output:
   
   ```json
   {
     "expiration": "2027-10-10T07:00:00Z",
     "licenseType": "Offline",
     "maxEngines": 10,
     "scanningEnabled": true,

>      "subject": "sf|0014W0000341gYFQAY|a2X4W000000ZeatUAC",

Reviewer: ```suggestion
     "subject": "example",
```

let's not expose the ugly dev's subject.
